### PR TITLE
Add test: Verify new node pool lands on given set of availability zones

### PR DESCRIPTION
**Pull request by tuommaki**
When creating multi-AZ node pool, verify that created VM node pool actually uses configured availability zones.

### CODE REVIEW

File: giantswarm-plugin.yaml [hunk 1]
```diff
@@ -9,6 +9,26 @@ spec:
     - name: TC_KUBECONFIG
     - name: CP_KUBECONFIG
     - name: CLUSTER_ID
+    - name: AZURE_CLIENT_ID
```
Comment (whites11) [inline giantswarm-plugin.yaml:12]: didn't we get this data from the secret?

**Review by whites11 — APPROVED** [review]

File: giantswarm-plugin.yaml (hunk 1 above)
Comment (tuommaki) [reply giantswarm-plugin.yaml:12]: Oh! VVery good point. I'll zap it!

Comment (whites11) [reply giantswarm-plugin.yaml]: are you sure "zap" is an accepted word for our new code of conduct? :P

Comment (tuommaki) [reply giantswarm-plugin.yaml]: With the claimed meaning of _to get rid of as if with sudden force_, in this context it must be :joy:

**Pull request merged**